Add "Relating Site Speed and Business Metrics" article (#3341)
GoogleChrome/web.dev: src/site/content/en/blog/speed-to-business/index.md

> ---

Reviewer: Please update the URL to this: `https://web.dev/site-speed-and-business-metrics`

layout: post
title: Relating Site Speed and Business Metrics
subhead: Leverage A/B testing to evaluate the impact of site speed on your business metrics
authors:
  - msdikla
  - martinschierle
  - bpj 

> date: 2020-06-25

Reviewer: ```suggestion
date: 2020-07-07
```

> hero: hero.jpg

Reviewer: How about this for the hero:

![louis-reed-pwcKF7L4-no-unsplash](https://user-images.githubusercontent.com/4713486/86873180-ad17d580-c092-11ea-8a83-1cddb94dfa25.jpg)

## Step 1: Pick a page to A/B test

Your aim is to test the hypothesis that page speed relates to your business metrics. For simplicity's sake, you can initially limit yourself to identifying a single page for analysis. Future work can expand to multiple pages of the same type to verify findings, and then to other areas of the site. Some suggestions for where to start are provided at the bottom of this step. Several requirements drive the page selection process:

*   The A/B test should only be run on the devices of mobile users. Globally, the partner sites we assist see an average of [more than 50%](https://gs.statcounter.com/platform-market-share/desktop-mobile-tablet/worldwide) (and growing!) of their traffic coming from mobile, but this can increase significantly based on [region](https://gs.statcounter.com/platform-market-share/desktop-mobile-tablet/india) and [vertical](https://www.salecycle.com/blog/stats/2019-ecommerce-year-in-review-infographic/). Mobile devices are more sensitive to slower websites due to processing and memory constraints and less stable networks. Also, on-the-go usage patterns mean expectations for speed are higher.

> *   The page you choose for testing should be an important part of your conversion funnel. 
> 
>     Every site has a different purpose, so every site tracks different success metrics. These metrics are usually related to a user journey which is analyzed using a funnel. For example, users on an e-commerce website will have to navigate through a home page, category pages, product pages, and a checkout page to complete a purchase. If you're optimizing for conversions, one of those pages would be a good candidate. 

Reviewer: ```suggestion
*   The page you choose for testing should be an important part of your conversion funnel. 
    Every site has a different purpose, so every site tracks different success metrics. These metrics are usually related to a user journey which is analyzed using a funnel. For example, users on an e-commerce website will have to navigate through a home page, category pages, product pages, and a checkout page to complete a purchase. If you're optimizing for conversions, one of those pages would be a good candidate. 
```

> *   The page should have a singular purpose. Unless your site has a very specific mission, it's usually best to avoid the temptation to select the homepage for your test. Many commercial sites' homepages are portals to a wide variety of functionality that will make your analysis noisy. For example, if you're optimizing for pageviews per session on a news site, it might be best to exclude the non-commercial portions of the site and focus on monetized sections and articles.

Reviewer: ```suggestion
*   The page should have a singular purpose. Unless your site has a very specific mission, it's usually best to avoid using the homepage for your test. Many commercial sites' homepages are portals to a wide variety of functionality that will make your analysis noisy. For example, if you're optimizing for pageviews per session on a news site, it might be best to exclude the non-commercial portions of the site and focus on monetized sections and articles.
```

*   The page chosen should get enough traffic so that you don't have to wait long before getting a statistically significant result.

> *   The selected page should be relatively slow. In fact, the slower the better. This not only means that you will likely have an easier time improving the page, it also means that the data should be much clearer. You can do a quick scan through your [Google Analytics Speed Report](https://support.google.com/analytics/answer/1205784?hl=en) to see which of your pages are slowest. 

Reviewer: ```suggestion
*   The selected page should be relatively slow. In fact, the slower the better. This not only means that you will likely have an easier time improving the page, it also means that the data should be much clearer. You can do a quick scan through your [Google Analytics Speed Report](https://support.google.com/analytics/answer/1205784) or [Search Console Core Web Vitals report](https://support.google.com/webmasters/answer/9205520) to see which of your pages are slowest. 
```

A couple of things to keep in mind:

1.  Avoid making any changes to the UI or UX. Aside from one page being faster than the other, changes must be invisible to users.
1.  Measuring is also a key aspect of this stage. While developing, lab testing tools such as Lighthouse should be used to indicate the effect your changes have on performance. Keep in mind that changes to one metric can often influence another. Once the pages are live, stick to RUM for a more accurate evaluation.

Creating performance variants can be done in different ways. For the purpose of the test, you would want to do so as simply as possible. Below are a few options.

**Create a faster page**

*   Use a tool like [Squoosh](https://squoosh.app/) to manually optimize the images on your test page
*   [Use DevTools code coverage](https://developers.google.com/web/tools/chrome-devtools/coverage) to manually eliminate unused JavaScript or CSS just for that one page
*   [Efficiently load third-party scripts](/efficiently-load-third-party-javascript/)
*   Use a tool like [Critical](https://github.com/addyosmani/critical) to break out and inline critical CSS

> *   Remove non-critical JavaScript code that does not impact the user experience and which you can do without for the purpose of the test (for example, certain third party libraries)

Reviewer: ```suggestion
*   Remove non-critical JavaScript code that does not impact the user experience and which you can do without for the purpose of the test (for example, certain third-party libraries)
```

{% Details %}

{% DetailsSummary %}

Example

{% endDetailsSummary %}

Example of AB testing  performance changes on a given product detail page (PDP) via server-side testing:

![Server side testing diagram](test-server-side.jpg)

The request goes to the backend, which distributes the users to the two different versions of the page. While this is in general a good setup, it often needs IT resources to set up the server-side split.

Here is an example of a client side testing setup, using the prior page (the homepage in the diagram below) to run the testing JavaScript:

![Client side testing diagram](test-client-side.jpg)

> The testing JavaScript manipulates the outgoing link to give the two test groups of users links to the two versions of the PDP in question. This is easy to setup and maintain through common AB testing tools like Optimizely or Optimize and does not influence the performance test because the testing JavaScript runs on a different page.

Reviewer: ```suggestion
The testing JavaScript manipulates the outgoing link to give the two test groups of users links to the two versions of the PDP in question. This is easy to setup and maintain through common A/B testing tools like Optimizely or Optimize and does not influence the performance test because the testing JavaScript runs on a different page.
```
